### PR TITLE
Add driver generator commands

### PR DESCRIPTION
**Pull request by nronas**
Adds a code generator for add drivers.

Sample output:

Screenshot 2021-04-14 at 6 15 45 PM

### CODE REVIEW

File: commands/new.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	cmd.AddCommand(
+		NewDriverCmd(),
+	)
+
+	return cmd
+}
+
+func NewDriverCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:           "driver",
```
Comment (mgdelacroix) [inline commands/new.go]: I'd add the parameter there, like `driver <name>`

File: commands/new.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+}
+
+func newDriverCmdF(cmd *cobra.Command, args []string) error {
+	driverName := args[0]
+	morph.InfoLogger.Printf("Generating necessary file structure for %q driver\n", driverName)
+
+	if _, err := os.Stat(baseDriverPath + "/" + driverName); !os.IsNotExist(err) {
```
Comment (mgdelacroix) [inline commands/new.go]: we should use `filepath.Join` here for proper windows support

File: commands/new.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+	}
+
+	return generateNewDriver(driverName)
+}
+
+func generateNewDriver(driverName string) error {
+	morph.InfoLoggerLight.Printf("== CreateDriver: %q: generating driver files\n", driverName)
+	morph.InfoLoggerLight.Println("=================================================")
+
+	f := NewFile(driverName)
```
Comment (mgdelacroix) [inline commands/new.go:56]: if the only call to a base method of jen is this one, can we change the import to be a named one? (I mean, avoid the `import . "github.com/..."`)

File: commands/new.go [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+		"github.com/go-morph/morph/models":  "models",
+	})
+
+	f.Var().Add(Id("driverName")).Op("=").Add(Lit(driverName))
+	f.Var().Add(Id("defaultConfig")).Op("=").Add(Op("&")).Id("Config").Values(Dict{
+		Id("MigrationsTable"): Lit("schema_migrations"),
+	})
+	f.Comment("add here any custom driver configuration")
+	f.Var().Add(Id("configParams")).Op("=").Add(Index().String().Values())
+
```
Comment (mgdelacroix) [inline commands/new.go:71]: I don't know how precise/verbose we'd like to be with the `f.Line()` calls, but I'd add one after the imports and another one here to sepparate vars from the init func

File: commands/new.go [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+	f.Func().Id("init").Params().Block(
+		Qual("github.com/go-morph/morph/drivers", "Register").Call(Lit(driverName), Id("&"+driverName).Values()),
+	)
+
+	f.Line()
+
```
Comment (mgdelacroix) [inline commands/new.go]: based on the other `f.Line()` calls, that are directly on top of the next block of code, I'd remove this whiteline

File: commands/new.go [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+	f.Type().Id("Config").Struct(
+		Id("MigrationsTable").String(),
+		Comment("Add more properties here"),
+	)
+
+	f.Line()
+
```
Comment (mgdelacroix) [inline commands/new.go]: same here

File: commands/new.go [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+	f.Type().Id(driverName).Struct(
+		Id("config").Add(Id("*Config")),
+		Comment("Add more properties here"),
+	)
+
+	f.Line()
+
```
Comment (mgdelacroix) [inline commands/new.go]: same here

File: commands/new.go [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+	})
+
+	f.Line()
+	f.Func().Params(Id("driver").Id("*"+driverName)).Id("AppliedMigrations").Params().Params(Op("[]*").Qual("github.com/go-morph/morph/models", "Migration"), Error()).BlockFunc(func(g *Group) {
+		g.Comment("Implement the functionality that returns which migrations has already been applied in storage.")
+		g.Panic(Lit("implement me"))
+	})
+
+	driverDir := baseDriverPath + "/" + driverName
+	driverFile := driverDir + "/" + driverName + ".go"
```
Comment (mgdelacroix) [inline commands/new.go]: these two should use `filepath` as well

File: commands/new.go (hunk 3 above)
Comment (nronas) [reply commands/new.go:56]: unfortunately is not, all the `Id` calls are basically calls to Jen.

File: commands/new.go (hunk 4 above)
Comment (nronas) [reply commands/new.go:71]: Imports are automatically handled, so the top of the file looks good i think:

```go
// Initial code generated by generator.
package mysql

import (
	"github.com/go-morph/morph/drivers"
	"github.com/go-morph/morph/models"
)

var driverName = "mysql"
var defaultConfig = &Config{MigrationsTable: "schema_migrations"}

// add here any custom driver configuration
var configParams = []string{}

func init() {
	drivers.Register("mysql", &mysql{})
}

type Config struct {
	MigrationsTable string
	// Add more properties here
}

type mysql struct {
	config *Config
	// Add more properties here
}

func WithInstance(dbInstance interface{}, config *Config) (drivers.Driver, error) {
	return &mysql{config: config}, nil
}
```

File: commands/new.go (hunk 3 above)
Comment (mgdelacroix) [reply commands/new.go:56]: right, totally missed those. Happy to leave it as is then

File: commands/new.go (hunk 4 above)
Comment (mgdelacroix) [reply commands/new.go:71]: :heart_eyes:

File: commands/new.go [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+		RunE:          newDriverCmdF,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+}
+
+func newDriverCmdF(cmd *cobra.Command, args []string) error {
+	driverName := args[0]
+	morph.InfoLogger.Printf("Generating necessary file structure for %q driver\n", driverName)
+	driverDir := filepath.Join(baseDriverPath, "/", driverName)
```
Comment (mgdelacroix) [inline commands/new.go]: nit:

```suggestion
	driverDir := filepath.Join(baseDriverPath, driverName)
```

File: commands/new.go [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+	})
+
+	f.Line()
+	f.Func().Params(Id("driver").Id("*"+driverName)).Id("AppliedMigrations").Params().Params(Op("[]*").Qual("github.com/go-morph/morph/models", "Migration"), Error()).BlockFunc(func(g *Group) {
+		g.Comment("Implement the functionality that returns which migrations has already been applied in storage.")
+		g.Panic(Lit("implement me"))
+	})
+
+	driverDir := filepath.Join(baseDriverPath, "/", driverName)
+	driverFile := filepath.Join(driverDir, "/", driverName+".go")
```
Comment (mgdelacroix) [inline commands/new.go]:
```suggestion
	driverDir := filepath.Join(baseDriverPath, driverName)
	driverFile := filepath.Join(driverDir, driverName+".go")
```

**Review by mgdelacroix — DISMISSED** [review]
Just a small thing added. LGTM otherwise!! :+1:

File: commands/new.go (hunk 9 above)
Comment (nronas) [reply commands/new.go]: 🤦

**Review by mgdelacroix — APPROVED** [review]

**Pull request merged**